### PR TITLE
TST (string dtype): resolve all xfails in JSON IO tests

### PR DESCRIPTION
**Pull request by WillAyd**

### CODE REVIEW

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: pandas/tests/io/json/test_pandas.py [hunk 1]
```diff
@@ -710,6 +707,9 @@ def test_series_roundtrip_object(self, orient, dtype, object_series):
         if orient != "split":
             expected.name = None
 
+        if using_string_dtype():
+            expected = expected.astype(pd.StringDtype(na_value=np.nan))
```
[the remark this answers is not in the document]
Comment (WillAyd) [reply pandas/tests/io/json/test_pandas.py:711]: Good to know - I missed that on the prior SQL PR so will go back and do that too, so we have one consistent method internally

**Comment by jorisvandenbossche** [comment]
While looking at centralizing `to_pandas()` conversion, I noticed that for JSON we are not yet handling the case of the string dtype for the pyarrow engine (engine, not dtype_backend), so it's a bit surprising actually that all tests are passing here ...

File: pandas/tests/io/json/test_pandas.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply pandas/tests/io/json/test_pandas.py]: I think on the sql PR you wanted to construct the actual dtype instance? In that case the above is fine. I am mostly using `"str"` in the tests.

**Comment by WillAyd** [comment]
Interesting - I didn't even realize we had a pyarrow engine for JSON!

Is your expectation that those cases would be returning `dtype=object` and cause a failure?

**Comment by jorisvandenbossche** [comment]
> Is your expectation that those cases would be returning `dtype=object` and cause a failure?

Yes, at least if we have proper testing of it ..

**Comment by jorisvandenbossche** [comment]
But let's do that in a next PR, going to merge this one already

**Comment by WillAyd** [comment]
Alright - I'll take a look. I wonder if there's some post-processing in JSON that is masking that...

I'll take care of the backport as well. Thanks for merging

**Comment by jorisvandenbossche** [comment]
Manual backport -> https://github.com/pandas-dev/pandas/pull/60327

**Pull request merged**